qa/tasks/ses_qa.py

```diff
@@ -146,13 +146,16 @@ def drive_replace_initiate(self, **kwargs):
         In order to "hide" an existing disk from the ceph.c_v in teuthology
         the disk is formatted and mounted.
         """
-        # :TODO: make this random and find correct disk device
-        osd_id = "2"
+        from random import randint
+
         total_osds = self.master_remote.sh(number_of_osds_in_cluster)
+        osd_id = randint(0, int(total_osds) - 1)
         assert int(total_osds) == 4, "Unexpected number of osds {} (expected 4)".format(total_osds)
         self.master_remote.sh("sudo ceph osd tree --format json | tee before.json")
-        self.master_remote.sh("sudo salt-run osd.replace {} 2>/dev/null".format(osd_id))
-        self.master_remote.sh("sudo mkfs.ext4 -F /dev/vdd; sudo mount /dev/vdd /mnt")
+        osd_path = self.master_remote.sh("sudo salt '*' cephdisks.find_by_osd_id {} --out json | jq '.[][].path'".format(osd_id))
+        osd_path = osd_path.replace("\"", '').strip()
```

Comment:
instead of this, use "jq -r" in previous command, it will strip json output values.

Reply:
`jq -j` fits better here - https://stedolan.github.io/jq/manual/#Invokingjq

Reply:
> `jq -j` fits better here - https://stedolan.github.io/jq/manual/#Invokingjq

not quite true if you have multiple values, it is hard to separate them when using '-j' ;-)